paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -545,15 +544,10 @@ internal class PaymentSheetViewModelTest {
         val selection = PaymentSelection.Saved(PaymentMethodFixtures.CARD_PAYMENT_METHOD)
         viewModel.updateSelection(selection)
 
-        var stripeIntent: StripeIntent? = null
```

Comment:
Rather than remove turbine, I think it would be helpful to rewrite the test a little different.

```
@Test
    fun `onPaymentResult() with non-success outcome should report failure event`() = runTest {
        val viewModel = createViewModel(shouldFailLoad = true)
        val selection = PaymentSelection.Saved(PaymentMethodFixtures.CARD_PAYMENT_METHOD)
        viewModel.updateSelection(selection)

        viewModel.stripeIntent.test {
            viewModel.onPaymentResult(PaymentResult.Failed(Throwable()))
            verify(eventReporter).onPaymentFailure(selection)

            val stripeIntent = awaitItem()
            assertThat(stripeIntent).isNull()
        }
    }
```

Turbine does more assertions, which can be beneficial.

Reply:
By using turbines `.test` method. We ensure that exactly the events we expected are emitted. If we don't use turbine, we could get intermediate events that _might_ not be expected. But turbine guards against that.